src/selectors/entities/channels.js

```diff
@@ -416,36 +416,57 @@ export const getUnreads = createSelector(
     getMyChannelMemberships,
     getUsers,
     getCurrentUserId,
-    (channels, myMembers, users, currentUserId) => {
-        let messageCount = 0;
-        let mentionCount = 0;
+    getCurrentTeamId,
+    getMyTeams,
+    getTeamMemberships,
+    (channels, myMembers, users, currentUserId, currentTeamId, myTeams, myTeamMemberships) => {
+        let messageCountForCurrentTeam = 0; // Includes message count from channels of current team plus all GM'S and all DM's across teams
+        let mentionCountForCurrentTeam = 0; // Includes mention count from channels of current team plus all GM'S and all DM's across teams
         Object.keys(myMembers).forEach((channelId) => {
             const channel = channels[channelId];
             const m = myMembers[channelId];
 
-            if (channel && m) {
+            if (channel && m && (channel.team_id === currentTeamId || channel.type === 'D')) {
```

Comment:
Does this check need to also include group channels as well?

Also, maybe we could remove `channel.team_id === currentTeamId` and just get the unread count for the current team the same way that we do for the other teams in the loop below? I'm not sure if we need to treat the current team differently than the other teams.

Reply:
we have all those checks for removing archived channels and deleted users unread count to be in sync with what is in LHS. 

If we remove that and do the same for all teams then we will end up with regressions as it will cause unreads to go out of sync with teams with archived channels or unread users

we cannot fix the case for archived channels unread count for other teams here and the best way to fix it would be by doing it server side and hence my question if we should revert the above PR in my comment so we don't need this PR.

Reply:
> Does this check need to also include group channels as well?

Aah i think we should have GM's as well here

Reply:
Tests fails. I will have a look at this tomorrow if i can do this in any better way.

Reply:
What if you have an archived channel or deleted user DM on another team. Will that not cause the count to get messed up?

Since that's an edge case, I'd be fine taking this PR for now, but I want to make a ticket to at least test this more thoroughly. The tests for that selector are so closely tied to the other selector tests, so it's really hard to ensure this is tested properly in different cases.

Reply:
What if you have an archived channel or deleted user DM on another team. Will that not cause the count to get messed up?
Deleted users will work as they are independent of team but archived channels will cause issues
as i was sating before in other comments

> we cannot fix the case for archived channels unread count for other teams here and the best way to fix it would be by doing it server side